### ISSUE TITLE
Angular Distillation Loss 

### ISSUE THREAD

**Issue by greendream182** [deepinsight/insightface#1087]

**Comment by greendream182** [comment]
![image](https://user-images.githubusercontent.com/21553054/78754800-b49fe380-79aa-11ea-918c-c497868b1d43.png)

**Comment by greendream182** [comment]
![image](https://user-images.githubusercontent.com/21553054/78754898-da2ced00-79aa-11ea-8028-0fcfd2697bef.png)

**Comment by greendream182** [comment]
有哪位大佬试过Angular Distillation Loss ，比较过等式（4）和（5）两种方法？
个人认为arcface 和等式（4） 计算的loss都是与角余弦相似度有关，正常应该比等式（5）更好，

**Comment by ctgushiwei** [comment]
应该是（4）好些吧

**Comment by YjDai** [comment]
@greendream182 你好，你后来用这个蒸馏方法训练了吗？

**Comment by greendream182** [comment]
用了（4），效果还好

**Comment by greendream182** [comment]
@YjDai 用了（4），效果还好

**Comment by YjDai** [comment]
@greendream182 你好，可以分享一下你添加的损失函数的代码吗？

**Comment by qiu-pinggaizi** [comment]
We tried cosine similarity distillation with L2 normalization for face recognition and got good results:

```python
student_norm = F.normalize(student_emb, p=2, dim=1)
teacher_norm = F.normalize(teacher_emb, p=2, dim=1)
cos_sim = F.cosine_similarity(student_norm, teacher_norm, dim=1)
distill_loss = (1 - cos_sim).mean()
```

Key finding: applying L2 normalization before cosine similarity is essential. Without it, the loss scale is unstable and training doesn't converge properly. With R100 as teacher and mbf_v3_se (3.9M) as student, we achieved AgeDB-30 97.35% — better than the non-distilled baseline (97.22%) on harder benchmarks.

Full implementation: https://github.com/qiu-pinggaizi/arcface_torch (see `losses_distill.py`)
